autoemulate/emulators/neural_networks/mlp.py

```diff
@@ -0,0 +1,33 @@
+from typing import Tuple
+
+import torch
+from torch import nn
+
+from autoemulate.emulators.neural_networks.neural_networks import TorchModule, register
+
+
+@register("mlp")
+class MLPModule(TorchModule):
+    def __init__(
+        self,
+        input_size: int = None,
+        output_size: int = None,
+        random_state: int = None,
+        hidden_sizes: Tuple[int] = (100,),
+    ):
+        super(MLPModule, self).__init__(
+            module_name="mlp",
+            input_size=input_size,
+            output_size=output_size,
+            random_state=random_state,
+        )
+        modules = []
+        for hidden_size in hidden_sizes:
+            modules.append(nn.Linear(in_features=input_size, out_features=hidden_size))
+            modules.append(nn.ReLU())
+            input_size = hidden_size
+        modules.append(nn.Linear(in_features=input_size, out_features=output_size))
+        self.model = nn.Sequential(*modules)
```

Comment:
Yes, I think it make sense that the hyperparameter settings live within each `TorchModule`

Reply:
I have moved the `get_grid_params` method to `TorchModule`. A problem with this approach is that the `module` is not initialized upon creation of `NeuralNetTorch`, and is only initialized when we call `fit` for the first time. So if we call `grid_params = nn_torch_model.get_grid_params()` before a hyperparameter search, we will get an error due to `self.module_.get_grid_params` does not exists yet.

Reply:
We can either always initialize the `module` in `NeuralNetTorch.__init__`, which would fail some cases in the estimator test suite, or before we do hyperparameter search, we initialize the module on our own.

Reply:
Yes I see, so this currently fails when running:

```
em = AutoEmulate()  
em.setup(X, y, model_subset=["NeuralNetTorch"], param_search=True)
em.compare()
```

with `AttributeError: 'NeuralNetTorch' object has no attribute 'module_'`

My feeling is that initializing the `module` in `NeuralNetTorch.__init__` is fine and we just add the failed tests to `"_xfail_checks"`, because this seems like what skorch is intending to do anyway.